connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala

```diff
@@ -644,6 +645,34 @@ class ClientE2ETestSuite extends RemoteSparkSession with SQLHelper {
       .collect()
     assert(result sameElements expected)
   }
+
+  test("json from Dataset[String]") {
+    val session = spark
+    import session.implicits._
+    val expected = Seq(
+      new GenericRowWithSchema(
+        Array(73, "Shandong", "Kong"),
+        new StructType().add("age", LongType).add("city", StringType).add("name", StringType)))
+    val ds = Seq("""{"name":"Kong","age":73,"city":'Shandong'}""").toDS()
+    val result = spark.read.option("allowSingleQuotes", "true").json(ds)
+    checkSameResult(expected, result)
```

Comment:
https://github.com/apache/spark/blob/69dd20b5e45c7e3533efbfdc1974f59931c1b781/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L404-L417

From the code of the server side, `userSpecifiedSchema` is an `Option[StructType]` and default is None, so we can use it without specifying the`userSpecifiedSchema`?  Or is my test case not the correct scenario?

Reply:
Thanks ~

Reply:
Probably we should add the user provided schema in the message? Or always discard it?

Reply:
Will inferFromDataset trigger an job? If so, I think we’d better skip it if possible